server/monitor/src/main/java/org/apache/accumulo/monitor/rest/trace/TracesResource.java

```diff
@@ -0,0 +1,372 @@
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.IOException;
+import java.security.PrivilegedAction;
+import java.security.PrivilegedExceptionAction;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.TreeMap;
+
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.Scanner;
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.client.security.tokens.AuthenticationToken;
+import org.apache.accumulo.core.client.security.tokens.AuthenticationToken.Properties;
+import org.apache.accumulo.core.client.security.tokens.KerberosToken;
+import org.apache.accumulo.core.client.security.tokens.PasswordToken;
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.util.Pair;
+import org.apache.accumulo.monitor.Monitor;
+import org.apache.accumulo.server.client.HdfsZooInstance;
+import org.apache.accumulo.server.security.SecurityUtil;
+import org.apache.accumulo.tracer.SpanTree;
+import org.apache.accumulo.tracer.SpanTreeVisitor;
+import org.apache.accumulo.tracer.TraceDump;
+import org.apache.accumulo.tracer.TraceFormatter;
+import org.apache.accumulo.tracer.thrift.Annotation;
+import org.apache.accumulo.tracer.thrift.RemoteSpan;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.security.UserGroupInformation;
+
+/**
+ *
+ * Generates a list of traces with the summary, by type, and trace details
+ *
+ * @since 2.0.0
+ *
+ */
+@Path("/trace")
+@Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+public class TracesResource {
+
+  /**
+   * Generates a trace summary
+   *
+   * @param minutes
+   *          Range of minutes to filter traces
+   * @return Trace summary in specified range
+   */
+  @Path("summary/{minutes}")
+  @GET
+  public RecentTracesList getTraces(@DefaultValue("10") @PathParam("minutes") int minutes) throws Exception {
+
+    RecentTracesList recentTraces = new RecentTracesList();
+
+    Pair<Scanner,UserGroupInformation> pair = getScanner();
+    final Scanner scanner = pair.getFirst();
+    if (scanner == null) {
+      return recentTraces;
+    }
+
+    Range range = getRangeForTrace(minutes);
+    scanner.setRange(range);
+
+    final Map<String,RecentTracesInformation> summary = new TreeMap<>();
+    if (null != pair.getSecond()) {
+      pair.getSecond().doAs(new PrivilegedAction<Void>() {
```

Comment:
This is how the old Traces did it, as with other parts of the code I wrote (and now that I think of it all of it with few exceptions, such as the status report), I didn't change the original logic of getting the information from Accumulo, I just implemented an endpoint to expose the information through REST API.